docs/design.md

```diff
@@ -1,30 +1,28 @@
+Thanos is a set of components that can be composed into a highly available Prometheus setup with long term storage capabilities. Its main goals are operation simplicity and retaining of Prometheus's reliability properties.
 
-Unlike existing solutions it does not attempt to integrate with the official yet experiemental remote APIs. This has a variety of reasons, which go beyond the scope of this document.
-Instead its main integration point is the Prometheus 2.0 storage format ([spec][tsdb-format], [slides][tsdb-talk]) itself through the storage engine's [library][tsdb-lib].
+The Prometheus metric data model and the 2.0 storage format ([spec][tsdb-format], [slides][tsdb-talk]) are the foundational layers of all components in the system.
 
 ## Architecture
 
-### Shipping Data
+Thanos is a clustered system of components with distinct and decoupled purposes. Clustered components can be categorized as follows:
 
-```
-┌────────────┬─────────┐                   ┌────────────┬─────────┐
-│ Prometheus │ Sidecar │        ...        │ Prometheus │ Sidecar │
-└────────────┴────┬────┘                   └────────────┴────┬────┘
-                  │                                          │
-                Blocks                                     Blocks
-                  │                                          │
-                  v                                          v
-              ┌──────────────────────────────────────────────────┐
-              │               Google Cloud Storage               │
-              └──────────────────────────────────────────────────┘
-```
+* Data sources
+* Stores
+* Queriers
+
+### Data Sources
+
+A data source is a very generlized definition of a component that produces or collects metric data. Source advertise this data in the cluster to potential clients. Metric data can be retrieved via a well-known gRPC service.
 
-At the foundation of PromLTS stands the vanilla Prometheus server itself. Integration into the overall systems happens through a sidecar. No changes to the general setup should be necessary.
+Thanos provides two components that act as data sources: the Prometheus sidecar and rule nodes.
 
-Prometheus writes "blocks" of data, which represent all time series data collected during a fixed time window. A block is a simple directory with a few files:
+The sidecar implements the gRPC service on top of Prometheus' [HTTP and remote-read APIs][prom-http-api]. The rule node directly implements it on top of the Prometheus storage engine it is running.
+
+#### Data Backup
+
+Data sources that persist their data for long-term storage do so via the Prometheus 2.0 storage engine. The storage engine periodically produces immutable blocks of data for a fixed time range. A block is a directory with a handful of larger files containing all sample data and peristed indices that are required to retrieve the data:
```

Comment:
do they persist via Prometheus 2.0 storage engine or actually bucket syncing mechanism (shipper)?

Reply:
Persistence happens through Prometheus. The shipper really just does dumb uploading – now with some extra logic to not reupload etc.